src/ReactLeafletMap/layers/VectorLayer/index.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+
+import CanvasVectorLayer from "../CanvasVectorLayer"
+import DiagramLayer from "../DiagramLayer";
+import IndexedVectorLayer from "../IndexedVectorLayer";
+import SvgVectorLayer from "../SvgVectorLayer";
+import TiledVectorLayer from "../TiledVectorLayer";
+
+class VectorLayer extends React.PureComponent {
+	static propTypes = {
+		layerKey: PropTypes.string,
+		uniqueLayerKey: PropTypes.oneOfType([
+			PropTypes.number,
+			PropTypes.string
+		]),
+		onClick: PropTypes.func,
+		opacity: PropTypes.number,
+		options: PropTypes.object,
+		type: PropTypes.string,
+		view: PropTypes.object,
+		zoom: PropTypes.number,
+		zIndex: PropTypes.number
+	};
+
+	constructor(props) {
```

Comment:
it is not necessary, is not it?

Reply:
why?

Reply:
https://reactjs.org/docs/react-component.html#constructor

Reply:
632951677d08608be18ef31770d28b59e3aa468d